client/lib/route/add-query-args.js

```diff
@@ -12,6 +12,14 @@ export default function( args, url ) {
 		throw new Error( 'addQueryArgs expects the second argument to be a string.' );
 	}
 
+	// Remove any undefined query args
+	args = Object.keys( args ).reduce( ( defined, key ) => {
```

Comment:
👍 for `pickBy`

I think that changing `addQueryArgs` in this way is a good idea because:

1. I believe it to be more intuitive for people coming across it for the first time
2. It's covered by unit tests, and this change does not break them
3. I couldn't find any other use in calypso that it would harm
4. The empty `[]` technique is tricky and would probably cause confusion in the future